drv/lpc55-swd/src/main.rs

```diff
@@ -715,44 +707,34 @@ impl NotificationHandler for ServerImpl {
                     }
                 }
                 Err(e) => {
-                    // This should only fail if JTAG_DETECT or SP_RESET are currently asserted.
+                    // This should only fail if JTAG_DETECT or SP_RESET are
+                    // currently asserted.
                     ringbuf_entry!(Trace::TimerHandlerError(e));
                 }
             }
         }
 
         if (bits & notifications::SP_RESET_IRQ_MASK) != 0 {
-            ringbuf_entry!(Trace::SpResetFired);
-            if !invalidate && !self.do_handle_sp_reset() {
-                ringbuf_entry!(Trace::InvalidateSpMeasurement);
-                // Clear the attestation log
-                invalidate = true;
-            }
-            self.next_use_must_setup_swd();
-
-            //  We are not going to try to measure/trust the SP
-            //  when there is a glitch on the JTAG_DETECT signal.
+            //  We are not going to try to measure/trust the SP when there is a
+            //  glitch on the JTAG_DETECT signal.
             //
             //  e.g. JTAG_DETECT fired but before the handler was called, it
             //  de-asserted so that the SP_RESET that also fired could be
             //  handled successfully.
-            //}
+            ringbuf_entry!(Trace::SpResetFired);
+            if matches!(self.state, SwdState::Connected { .. })
+                && !self.do_handle_sp_reset()
+            {
+                // If handling the SP reset failed, clear the attestation log
+                ringbuf_entry!(Trace::InvalidateSpMeasurement);
+                self.invalidate_sp_measurement();
+            }
+            self.next_use_must_setup_swd();
 
+            // Squelch the interrupts generated by this handler's toggling of
+            // SP_RESET.
             const SLOT: PintSlot = ROT_TO_SP_RESET_L_IN_PINT_SLOT;
```

Comment:
I'd suggest inlining this constant since it appears to only be used in one place now.

Reply:
Done in f10cdc98e671